src/System.Threading.AccessControl/tests/ThreadingAclExtensionsTests.cs

```diff
@@ -45,5 +53,247 @@ public static void ExistenceTest_Unix()
             Assert.Throws<PlatformNotSupportedException>(() => m.GetAccessControl());
             Assert.Throws<PlatformNotSupportedException>(() => m.SetAccessControl(new MutexSecurity()));
         }
+
+        #endregion
+
+        #region EventWaitHandle
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void EventWaitHandle_Create_NullSecurity()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("eventSecurity", () =>
+            {
+                using EventWaitHandle eventHandle = EventWaitHandleAcl.Create(initialState: true, EventResetMode.AutoReset, "Test", out bool createdNew, eventSecurity: null);
+            });
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void EventWaitHandle_Create_NullName()
+        {
+             using EventWaitHandle eventHandle = EventWaitHandleAcl.Create(initialState: true, EventResetMode.AutoReset, null, out bool createdNew, GetBasicEventWaitHandleSecurity());
+             Assert.NotNull(eventHandle);
+             Assert.True(createdNew);
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void EventWaitHandle_Create_NullNameMultipleNew()
+        {
+            using EventWaitHandle eventHandle1 = EventWaitHandleAcl.Create(initialState: true, EventResetMode.AutoReset, null, out bool createdNew1, GetBasicEventWaitHandleSecurity());
+            Assert.NotNull(eventHandle1);
+            Assert.True(createdNew1);
+
+            using EventWaitHandle eventHandle2 = EventWaitHandleAcl.Create(initialState: true, EventResetMode.AutoReset, null, out bool createdNew2, GetBasicEventWaitHandleSecurity());
+            Assert.NotNull(eventHandle2);
+            Assert.True(createdNew2);
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void EventWaitHandle_Create_EmptyName()
+        {
+            using EventWaitHandle eventHandle = EventWaitHandleAcl.Create(initialState: true, EventResetMode.AutoReset, string.Empty, out bool createdNew, GetBasicEventWaitHandleSecurity());
+            Assert.NotNull(eventHandle);
+            Assert.True(createdNew);
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void EventWaitHandle_Create_EmptyNameMultipleNew()
```

Comment:
This matches the behavior of the existing `EventWaitHandle` constructors that take no security object: Two nameless EWH events created in a row will both be created new.

Reply:
@JeremyKuhne told me this may not be the correct behavior. I'm investigating.

Reply:
Nevermind, this is correct.